### PR TITLE
[docs] Correct the documentation on observability support with Native AOT on iOS and Mac Catalyst

### PR DESCRIPTION
**Pull request by ivanpovazan**
Addresses the comment: https://github.com/dotnet/docs-maui/pull/2614#discussion_r1838437503

/cc: @jkotas

### CODE REVIEW

**Comment by ivanpovazan** [comment]
@davidbritch please wait until we get an approval from @jkotas before merging in the PR

**Review by jkotas — APPROVED** [review]

**Review by davidbritch — APPROVED** [review]

**Pull request merged**
